xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearch.xml

```diff
@@ -198,14 +198,21 @@
     }
   };
 
-  var addFacetValueCheckbox = function() {
-    var checkBox = $(document.createElement('input')).attr('type', 'checkbox');
+  var addFacetValueCheckbox = function(index) {
+    // Create an id unique to the facet value.
+    let facetContainer = $(this).parents('.search-facet').first();
+    $(this).attr('id', facetContainer.attr('data-name') + '-' + index.toString());
+    // Initialize the checkbox.
+    let checkBox = $(document.createElement('input')).attr('type', 'checkbox');
+    checkBox.attr('aria-labelledby',$(this).attr('id') );
     checkBox.prop('checked', $(this).hasClass('selected'));
     // Add the 'checked' attribute so that it can be easily located with CSS.
     checkBox.prop('checked') &amp;&amp; checkBox.attr('checked', 'checked');
     checkBox.on('click', this.click.bind(this));
-    // Remove the 'selected' class because the selected state is marked using the check box.
-    $(this).removeClass('selected').before(checkBox);
+    // Remove the 'selected' class because the selected state is already marked using the check box.
+    $(this).removeClass('selected')
+    // Add the checkbox to the DOM
+    $(this).before(checkBox)
```

Comment:
This change doesn't seem to do anything. What about just reverting it?

Reply:
I think it helped made the code clearer, and it was close enough to code I modified here.

When I first went through the code, I couldn't spot at a glance where the element was added to the DOM, since this was done on the end of a line that was said to do something else.